payments/stripe/__init__.py

```diff
@@ -76,3 +94,141 @@ class StripeCardProvider(StripeProvider):
     """
 
     form_class = PaymentForm
+
+
+@dataclass
+class StripeProductData:
+    name: str
+    description: Optional[str] = field(init=False, repr=False, default=None)
+    images: Optional[str] = field(init=False, repr=False, default=None)
+    metadata: Optional[dict] = field(init=False, repr=False, default=None)
+    tax_code: Optional[str] = field(init=False, repr=False, default=None)
+
+
+@dataclass
+class StripePriceData:
+    currency: str
+    product_data: StripeProductData
+    unit_amount: int
+    recurring: Optional[dict] = field(init=False, repr=False, default=None)
+    tax_behavior: Optional[str] = field(init=False, repr=False, default=None)
+
+
+@dataclass
+class StripeLineItem:
+    price_data: StripePriceData
+    quantity: int
+    adjustable_quantity: Optional[dict] = field(init=False, repr=False, default=None)
+    dynamic_tax_rates: Optional[dict] = field(init=False, repr=False, default=None)
+    tax_rates: Optional[str] = field(init=False, repr=False, default=None)
+
+
+class StripeProviderV3(BasicProvider):
+    """Provider backend using `Stripe <https://stripe.com/>` api version 3_.
+
+    This backend does not support fraud detection.
+
+    :param api_key: Secret key assigned by Stripe.
+    :param payment_method_types: From Stripe API, comma separated
+    :param use_token: Use instance.token instead of instance.pk in client_reference_id
+    """
+
+    form_class = PaymentFormV3
+
+    def __init__(
+        self,
+        api_key,
+        use_token=True,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.api_key = api_key
+        self.use_token = use_token
+
+    def get_form(self, payment, data=None):
+        if not payment.transaction_id:
+            try:
+                session = self.create_session(payment)
+            except PaymentError as pe:
+                payment.change_status(PaymentStatus.ERROR, str(pe))
+                raise PaymentError(pe)
+            else:
+                payment.attrs.session = session
+                payment.transaction_id = session.get("id", None)
+                payment.save()
+
+        if "url" not in payment.attrs.session:
+            raise PaymentError(_("No `url` in payment.attrs.session"))
```

Comment:
The error doesn't make it very obvious what went wrong:

```suggestion
            raise PaymentError(_("Stripe returned a session without a URL"))
```

Are translations needed for this one?